torch/fx/proxy.py

```diff
@@ -153,6 +156,10 @@ def __bool__(self) -> bool:
     def keys(self):
         return self.tracer.keys(self)
 
+    def __len__(self):
+        warnings.warn("'builtins.len' is not supported. Replace `builtins.len` with 'torch.fx.len' and try again.")
+        return
```

Comment:
Instead of warning, this should throw a `RuntimeError`. As it is, this function will return `None`, and I think it will run into the same hard-to-interpret "can't be interpreted as an int" error. Can you make this change and add a test case for the exception case (using `self.assertRaisesRegex` in the test)?